symex-misc.el

```diff
@@ -218,78 +153,24 @@ executing it."
-           (symex-eval-thunk-elisp))
-          ((equal major-mode 'scheme-mode)
-           (symex-eval-thunk-scheme))
-          ((member major-mode symex-clojure-modes)
-           (symex-eval-thunk-clojure))
-          ((member major-mode symex-common-lisp-modes)
-           (symex-eval-thunk-common-lisp))
-          ((equal major-mode 'arc-mode)
-           (symex-eval-thunk-arc))
-          ((equal major-mode 'fennel-mode)
-           (symex-eval-thunk-fennel))
-          (t (error "Symex mode: Lisp flavor not recognized!")))))
+    (funcall (symex-interface-get-method :eval-thunk))))
 
 (defun symex-describe ()
   "Lookup doc on symex."
   (interactive)
   (save-excursion
     (forward-sexp)  ; selected symexes will have the cursor on the starting paren
-    (cond ((member major-mode symex-racket-modes)
-           (symex-describe-symbol-racket))
-          ((member major-mode symex-elisp-modes)
-           (symex-describe-symbol-elisp))
-          ((equal major-mode 'scheme-mode)
-           (symex-describe-symbol-scheme))
-          ((member major-mode symex-clojure-modes)
-           (symex-describe-symbol-clojure))
-          ((member major-mode symex-common-lisp-modes)
-           (symex-describe-symbol-common-lisp))
-          ((equal major-mode 'arc-mode)
-           (symex-describe-symbol-arc))
-          ((equal major-mode 'fennel-mode)
-           (symex-describe-symbol-fennel))
-          (t (error "Symex mode: Lisp flavor not recognized!")))))
+    (funcall (symex-interface-get-method :describe-symbol))))
 
 (defun symex-repl ()
   "Go to REPL."
   (interactive)
-  (cond ((member major-mode symex-racket-modes)
-         (symex-repl-racket))
-        ((member major-mode symex-elisp-modes)
-         (symex-repl-elisp))
-        ((equal major-mode 'scheme-mode)
-         (symex-repl-scheme))
-        ((member major-mode symex-clojure-modes)
-         (symex-repl-clojure))
-        ((member major-mode symex-common-lisp-modes)
-         (symex-repl-common-lisp))
-        ((equal major-mode 'arc-mode)
-         (symex-repl-arc))
-        ((equal major-mode 'fennel-mode)
-         (symex-repl-fennel))
-        (t (error "Symex mode: Lisp flavor not recognized!"))))
+  (funcall (symex-interface-get-method :repl)))
 
 (defun symex-run ()
   "Send to REPL."
   (interactive)
-  (cond ((member major-mode symex-racket-modes)
-         (symex-run-racket))
-        ((member major-mode symex-elisp-modes)
-         (symex-run-elisp))
-        ((equal major-mode 'scheme-mode)
-         (symex-run-scheme))
-        ((member major-mode symex-clojure-modes)
-         (symex-run-clojure))
-        ((member major-mode symex-common-lisp-modes)
-         (symex-run-common-lisp))
-        ((equal major-mode 'fennel-mode)
-         (symex-run-fennel))
-        (t (error "Symex mode: Lisp flavor not recognized!"))))
+  (funcall (symex-interface-get-method :run)))
 
 (cl-defun symex--new-scratch-buffer (buffer-name)
   "Create a new empty buffer.
```

Comment:
re: `switch-to-scratch-buffer`, we could include that in the new interface pattern by defining `symex-switch-to-scratch-buffer` as:

```
(defun symex-switch-to-scratch-buffer ()
  "Switch to scratch buffer."
  (interactive)
  (funcall (symex-interface-get-method :switch-to-scratch-buffer)))
```

And then each Lisp interface would define the function as something like:
```
(defun symex-racket-switch-to-scratch-buffer ()
  "Switch to scratch buffer."
  (let* ((buffer-name "*scratch - Racket*")
         (buf (or (get-buffer buffer-name)
                  (symex--new-scratch-buffer buffer-name))))
    (switch-to-buffer buf)))
```

We would need to move `symex--new-scratch-buffer` into a low-level utilities module, e.g. `symex-utils.el`, and then `(require 'symex-utils)` that in each of the `interface-` modules.